Source/WebCore/platform/graphics/gpu/cocoa/WebGPUSurfaceCocoa.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "IOSurface.h"
+#include "WebGPUSurface.h"
+#include "pal/graphics/WebGPU/WebGPUCanvasConfiguration.h"
+
+namespace WebCore {
+
+class WebGPUSurfaceCocoa final: public WebGPUSurface {
+public:
+    static RefPtr<WebGPUSurfaceCocoa> create(IntSize, PAL::WebGPU::CanvasConfiguration);
+
+    ~WebGPUSurfaceCocoa() = default;
+
+    // GraphicsLayerContentsDisplayDelegate overrides:
+    void prepareToDelegateDisplay(PlatformCALayer&) override;
+    void display(PlatformCALayer&) override;
+    GraphicsLayer::CompositingCoordinatesOrientation orientation() const override;
+
+    // WebGPUSurface overrides:
+    Ref<PAL::WebGPU::Texture> currentTexture() override;
+    PixelFormat pixelFormat() const override;
+    DestinationColorSpace colorSpace() const override;
```

Comment:
Nit: `final`

Reply:
it's already a final class: `class WebGPUSurfaceCocoa final: public WebGPUSurface {`

do we place final on the member function for redundancy / in case the class is changed from no longer being final? Since final on a member function of a final class does not change anything as far as I know, right?

Reply:
Oh, yeah, that makes sense. https://webkit.org/code-style-guidelines/#override-methods seems a little ambiguous here. Maybe this is worth a webkit-dev email about style?